sdk/provisioning/Azure.Provisioning/src/ModuleConstruct.cs

```diff
@@ -182,18 +182,36 @@ internal void WriteOutputs(MemoryStream stream)
             GetAllOutputsRecursive(this, outputsToWrite, false);
             foreach (var output in outputsToWrite)
             {
-                string value;
-                if (output.IsLiteral || ReferenceEquals(this, output.Resource.ModuleScope))
-                {
-                    value = output.IsLiteral ? $"'{output.Value}'" : output.Value;
-                }
-                else
+                stream.WriteLine(GetOutputString(output));
+            }
+        }
+
+        private string GetOutputString(Output output)
+        {
+            string value;
+
+            if (output.IsLiteral)
+            {
+                value = output.Kind switch
                 {
-                    value = $"{output.Resource.ModuleScope!.Name}.outputs.{output.Name}";
-                }
-                string name = output.Name;
-                stream.WriteLine($"output {name} string = {value}");
+                    BicepKind.Bool => output.Value.ToLower(),
+                    BicepKind.Int => output.Value,
+                    BicepKind.Array => output.Value,
+                    BicepKind.Object => output.Value,
+                    BicepKind.String => $"'{output.Value}'",
```

Comment:
Is `output.Value` already escaped if it contains a quote?

Reply:
I think the output.IsLiteral logic isn't needed. In fact, the method the AddOutput method shouldn't have an isLiteral parameter as it requires specifying a propertySelector so it would never be a literal. Literal outputs aren't really supported currently. I will remove this logic.